ReSharper.FSharp/src/FSharp.Psi/src/CodeFormatter/FSharpCodeFormatter.cs

```diff
@@ -0,0 +1,137 @@
+        return MinimalSeparatorType.NotRequired;
+
+      if (leftTokenType.IsWhitespace || rightTokenType.IsWhitespace)
+        return MinimalSeparatorType.NotRequired;
+
+      if (glueingCache.GetOrAdd(new FormatterImplHelper.TokenTypePair(leftTokenType, rightTokenType), AreTokensGlued))
+      {
+        return leftTokenType == FSharpTokenType.LINE_COMMENT
+          ? MinimalSeparatorType.NewLine
+          : MinimalSeparatorType.Space;
+      }
+
+      return MinimalSeparatorType.NotRequired;
+    }
+
+    public override bool IsNewLine(ITreeNode ws) => ws is NewLine;
+
+    public override ITreeNode CreateSpace(string indent, ITreeNode replacedSpace) => new Whitespace(indent);
+
+    public override ITreeNode CreateNewLine(LineEnding lineEnding, NodeType lineBreakType = null) =>
+      new NewLine(lineEnding.GetPresentation());
+
+    public override ITreeRange Format(ITreeNode firstElement,
+      ITreeNode lastElement,
+      CodeFormatProfile profile,
+      AdditionalFormatterParameters parameters = null)
+    {
+      parameters ??= AdditionalFormatterParameters.Empty;
+      var task = new FormatTask(firstElement, lastElement, profile);
+      task.Adjust(this);
+      if (task.FirstElement == null)
+        return new TreeRange(firstElement, lastElement);
+
+      var solution = task.FirstElement.GetSolution();
+      if (!solution.RdFSharpModel().EnableExperimentalFeaturesSafe())
+        return new TreeRange(firstElement, lastElement);
+
+      var formatterSettings = GetFormattingSettings(task.FirstElement, parameters);
+
+      DoDeclarativeFormat(
+        formatterSettings,
+        formatterInfoProvider,
+        null,
+        new[] {task},
+        parameters,
+        null,
+        null,
+        null,
+        false);
+
+      return new TreeRange(firstElement, lastElement);
+    }
+
+    public override void FormatInsertedNodes(ITreeNode nodeFirst, ITreeNode nodeLast, bool formatSurround) =>
+      FormatterImplHelper.FormatInsertedNodesHelper(this, nodeFirst, nodeLast, formatSurround);
+
+    public override ITreeRange FormatInsertedRange(ITreeNode nodeFirst, ITreeNode nodeLast, ITreeRange origin) =>
+      FormatterImplHelper.FormatInsertedRangeHelper(this, nodeFirst, nodeLast, origin, true);
+
+    public override void FormatReplacedNode(ITreeNode oldNode, ITreeNode newNode)
+    {
+      FormatInsertedNodes(newNode, newNode, true);
+
+      FormatterImplHelper.CheckForMinimumSeparator(this, newNode);
+    }
+
+    public override void FormatReplacedRange(ITreeNode first, ITreeNode last, ITreeRange oldNodes)
+    {
+      FormatInsertedNodes(first, last, false);
+
+      FormatterImplHelper.CheckForMinimumSeparator(this, first, last);
+    }
+
+    public override void FormatDeletedNodes(ITreeNode parent, ITreeNode prevNode, ITreeNode nextNode)
+      => FormatterImplHelper.FormatDeletedNodesHelper(this, parent, prevNode, nextNode, false);
+
+    public override string OverridenSettingPrefix => "// @formatter:";
```

Comment:
Does this string actually work?

Reply:
Yes, if setting has `EditorConfigEntryAlias` attribute, it's value can be overridden using this string.

Reply:
EditorConfigEntryAlias is not necessary really. Every setting in settings key marked with EditorConfigKey attribute could be here